src/aframe-components/scene-timer.js

```diff
@@ -0,0 +1,277 @@
+/* global AFRAME, STREET */
+
+/**
+ * Scene Timer Component
+ * A global timer component that manages time for the entire scene.
+ * Provides play, pause, stop functionality and timing information.
+ */
+AFRAME.registerComponent('scene-timer', {
+  schema: {
+    autoStart: { type: 'boolean', default: false },
+    format: { type: 'string', default: 'mm:ss:ff' } // mm:ss:ff, ss:ms, or raw
+  },
+
+  init: function () {
+    // Initialize timer state
+    this.elapsedTime = 0; // Time in milliseconds
+    this.startTime = null;
+    this.timerActive = false;
+    this.isPaused = false;
+    this.frameRate = 30; // Assumed frame rate for frame count
+
+    // Bind event handlers once to reuse the same function references
+    this.boundStartTimer = this.startTimer.bind(this);
+    this.boundPauseTimer = this.pauseTimer.bind(this);
+    this.boundStopTimer = this.stopTimer.bind(this);
+    this.boundReset = this.reset.bind(this);
```

Comment:
You can just use the same name, important if you remove the global STREET.timer usage.

Reply:
thanks 31f0212ba0e73d117a9bddd4e680c63c0203ebde